src/Microsoft.AspNet.OData.Shared/Query/DefaultSkipTokenHandler.cs

```diff
@@ -0,0 +1,387 @@
+                }
+
+                skipTokenGenerator = (obj) =>
+                {
+                    return GenerateSkipTokenValue(obj, model, orderByNodes);
+                };
+            }
+            return context.InternalRequest.GetNextPageLink(pageSize, instance, skipTokenGenerator);
+        }
+
+        /// <summary>
+        /// Returns a function that converts an object to a skiptoken value string
+        /// </summary>
+        /// <param name="lastMember"> Object based on which SkipToken value will be generated.</param>
+        /// <param name="model">The edm model.</param>
+        /// <param name="orderByNodes">QueryOption </param>
+        /// <returns></returns>
+        protected virtual string GenerateSkipTokenValue(Object lastMember, IEdmModel model, IList<OrderByNode> orderByNodes)
+        {
+            object value;
+            if (lastMember == null)
+            {
+                return String.Empty;
+            }
+            IEnumerable<IEdmProperty> propertiesForSkipToken = GetPropertiesForSkipToken(lastMember, model, orderByNodes);
+
+            String skipTokenvalue = String.Empty;
+            if (propertiesForSkipToken == null)
+            {
+                return skipTokenvalue;
+            }
+
+            int count = 0;
+            int lastIndex = propertiesForSkipToken.Count() - 1;
+            foreach (IEdmProperty property in propertiesForSkipToken)
+            {
+                bool islast = count == lastIndex;
+                string propertyName = property.Name;
+                IEdmStructuredObject obj = lastMember as IEdmStructuredObject;
+                if (obj != null)
+                {
+                    obj.TryGetPropertyValue(property.Name, out value);
+                }
+                else
+                {
+                    value = lastMember.GetType().GetProperty(property.Name).GetValue(lastMember);
+                }
+
+                String uriLiteral = String.Empty;
+                if (value == null)
+                {
+                    uriLiteral = ODataUriUtils.ConvertToUriLiteral(value, ODataVersion.V401);
+                }
+                else if (TypeHelper.IsEnum(value.GetType()))
+                {
+                    ODataEnumValue enumValue = new ODataEnumValue(value.ToString(), value.GetType().FullName);
+                    uriLiteral = ODataUriUtils.ConvertToUriLiteral(enumValue, ODataVersion.V401, model);
+                    uriLiteral = "'enumType'" + uriLiteral;
+
+                    //Use CLR property name for Enums
+                    propertyName = EdmLibHelpers.GetClrPropertyName(property, model);
+                }
+                else
+                {
+                    uriLiteral = ODataUriUtils.ConvertToUriLiteral(value, ODataVersion.V401, model);
+                }
+                skipTokenvalue += propertyName + propertyDelimiter + uriLiteral + (islast ? String.Empty : CommaDelimiter.ToString());
```

Comment:
> skipTokenvalue += propertyName + propertyDelimiter + uriLiteral + (islast ? String.Empty : CommaDelimiter.ToString()); [](start = 15, length = 119)

This pattern will generate a lot of internal string object. 
Maybe use a StringBuilder or a list to improve